spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -158,6 +178,134 @@ public Filter[] pushedFilters() {
     return pushedFilters;
   }
 
+  @Override
+  public boolean pushAggregation(Aggregation aggregation) {
+    if (!canPushDownAggregation(aggregation)) {
+      return false;
+    }
+
+    AggregateEvaluator aggregateEvaluator;
+    try {
+      List<Expression> aggregates =
+          Arrays.stream(aggregation.aggregateExpressions())
+              .map(agg -> SparkAggregates.convert(agg))
+              .collect(Collectors.toList());
+      aggregateEvaluator = AggregateEvaluator.create(schema, aggregates);
+    } catch (UnsupportedOperationException | IllegalArgumentException e) {
+      LOG.info("Skipped aggregate pushdown: " + e);
```

Comment:
I still find this awkward because it isn't telling us what failed directly. It also includes the message from the error without including the stack trace, which is a sign that the way we're passing information through an exception is for convenience and not because it is an exceptional case. I think that this is okay, but I'd really prefer to attempt to convert each aggregate and fail individually if we can't.

Reply:
I changed this to follow the `pushFilters(Filter[] filters)` style to convert aggregate one by one